src/action_request.cc

```diff
@@ -113,6 +113,7 @@ void ActionRequest::update()
             UpnpActionRequest_set_ActionResult(upnp_request, result);
             UpnpActionRequest_set_ErrCode(upnp_request, errCode);
         }
+        ixmlDocument_free(result);
```

Comment:
IMHO this leaves the `result` inside `upnp_request` as dirty pointer. Don't know how to solve this at the moment.